cdc/capture/capture.go

```diff
@@ -465,35 +474,25 @@ func (c *captureImpl) campaignOwner(ctx cdcContext.Context) error {
 			ownerFlushInterval, util.RoleOwner.String())
 		c.owner.AsyncStop()
 		c.setOwner(nil)
-
-		// if owner exits, resign the owner key,
-		// use a new context to prevent the context from being cancelled.
-		resignCtx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
-		if resignErr := c.resign(resignCtx); resignErr != nil {
-			if errors.Cause(resignErr) != context.DeadlineExceeded {
-				log.Info("owner resign failed", zap.String("captureID", c.info.ID),
-					zap.Error(resignErr), zap.Int64("ownerRev", ownerRev))
-				cancel()
-				return errors.Trace(resignErr)
-			}
-
-			log.Warn("owner resign timeout", zap.String("captureID", c.info.ID),
-				zap.Error(resignErr), zap.Int64("ownerRev", ownerRev))
-		}
-		cancel()
-
-		log.Info("owner resigned successfully",
-			zap.String("captureID", c.info.ID), zap.Int64("ownerRev", ownerRev))
+		// when owner exit, we don't `resign` because we will call `resign`
+		// at `AsyncClose`
 		if err != nil {
 			log.Warn("run owner exited with error",
 				zap.String("captureID", c.info.ID), zap.Int64("ownerRev", ownerRev),
 				zap.Error(err))
 			// for errors, return error and let capture exits or restart
 			return errors.Trace(err)
 		}
+
 		// if owner exits normally, continue the campaign loop and try to election owner again
 		log.Info("run owner exited normally",
 			zap.String("captureID", c.info.ID), zap.Int64("ownerRev", ownerRev))
+
+		// before a new cycle starts, we need resign and just need ignore error
```

Comment:
Please add more comment about why we should just ignore this error.

Reply:
the best way should return error, thanks!